map/light_framework.hpp

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "map/user.hpp"
+
+#include "ugc/storage.hpp"
+
+#include "base/assert.hpp"
+
+namespace lightweight
+{
+struct LightFrameworkTest;
+
+enum RequestType
+{
+  Empty = 0u,
+  NumberOfUnsentUGC = 1u << 0,
+  UserAuthStatus = 1u << 1,
+  NumberOfUnsentEdits = 1u << 2,
+  All = NumberOfUnsentUGC | UserAuthStatus | NumberOfUnsentEdits
+};
+
+using RequestTypeMask = unsigned;
+
+class Framework
+{
+public:
+  friend struct LightFrameworkTest;
+
+  Framework(RequestTypeMask request) : m_request(request)
+  {
+    CHECK_NOT_EQUAL(request, Empty, ("Mask is empty"));
+    CHECK_LESS_OR_EQUAL(request, All, ("Incorrect mask type:", request));
+
+    if (request & NumberOfUnsentUGC)
+    {
+      m_numberOfUnsentUGC = GetNumberOfUnsentUGC();
+      request ^= NumberOfUnsentUGC;
+    }
+
+    if (request & UserAuthStatus)
+    {
+      m_userAuthStatus = IsUserAuthenticated();
+      request ^= UserAuthStatus;
+    }
+
+    if (request & NumberOfUnsentEdits)
+    {
+      // TODO: Hasn't implemented yet.
+      request ^= NumberOfUnsentEdits;
+    }
+
+    CHECK_EQUAL(request, Empty, ("Incorrect mask type:", request));
+  }
+
+  template <RequestTypeMask Type>
+  auto Get() const;
+
+private:
+  RequestTypeMask m_request;
```

Comment:
= 0?

Reply:
Ok, but have you noticed there is only one ctor which is initializing this field?

Reply:
sure but I don't think it'll hurt